ed25519/ed25519_verify/src/ed25519_verify.rs

```diff
@@ -0,0 +1,83 @@
+extern crate ed25519_dalek;
+extern crate sha2;
+
+use ed25519_dalek::{PublicKey, Signature};
+use sha2::Sha512;
+
+#[derive(Debug)]
+pub struct Error(pub &'static str);
+
+impl From<&'static str> for self::Error {
+    fn from(val: &'static str) -> Self {
+        self::Error(val)
+    }
+}
+
+pub fn verify(input: &[u8]) -> Result<bool, self::Error> {
```

Comment:
I think it is better to specify the slice size here to enforce size checks on the caller size. Currently, if `verify` receives a short slice, it will panic.

Reply:
Okay.  So then I have to copy the 128byte slice from the calldata into a fixed size array.  Is there any way to get around this overhead?

Reply:
The next version of `ewasm-rust-api` will support slices, that will remove one level of buffering.